### PR TITLE
chore(docs): add doc for the test server

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Document briefly the tests servers, and refactor to split and reuse the express logic.

### CODE REVIEW

File: scripts/cli/index.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { buildClients } from '../buildClients.js';
 import { LANGUAGES, setVerbose } from '../common.js';
 import { ctsGenerateMany } from '../cts/generate.js';
 import { runCts } from '../cts/runCts.js';
-import { startTestServer } from '../cts/testServer';
+import { startTestServer } from '../cts/testServer/index.js';
```
Comment (shortcuts) [inline scripts/cli/index.ts:8]: is it required with esm to define the index file? 🤔

File: scripts/cts/testServer/timeout.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { setupServer } from '.';
+
+let timeoutCounter = 0;
+
+export function getTimeoutCounter(): number {
+  return timeoutCounter;
+}
+
+function addRoutes(app: express.Express): void {
+  app.get('/1/test/retry', (req, res) => {
```
Comment (shortcuts) [inline scripts/cts/testServer/timeout.ts:14]: this one is defined in scripts/cts/testServer/gzip.ts also, is it expected?

File: scripts/cts/testServer/algoliaMock.ts
Comment (shortcuts) [inline scripts/cts/testServer/algoliaMock.ts:1]: maybe rename it to replaceAllObjects explicitly so we will have 1 file per use case later on?

**Review by shortcuts — COMMENTED** [review]
noice

File: scripts/cli/index.ts (hunk 1 above)
Comment (millotp) [reply scripts/cli/index.ts:8]: Apparently not !

File: scripts/cts/testServer/timeout.ts (hunk 2 above)
Comment (millotp) [reply scripts/cts/testServer/timeout.ts:14]: yes, so that the retry strat can call the same endpoint but with different behavior

Comment (millotp) [reply scripts/cts/testServer/timeout.ts:14]: I'll add a comment

**Review by shortcuts — APPROVED** [review]
awesome

**Pull request merged**